pilot/pkg/xds/discovery.go

```diff
@@ -508,3 +508,36 @@ func (s *DiscoveryServer) Clients() []*Connection {
 	}
 	return clients
 }
+
+// SendResponse will immediately send the response to all connections.
+// TODO: additional filters can be added, for example namespace.
+func (s *DiscoveryServer) SendResponse(res *discovery.DiscoveryResponse) {
+	pending := []*Connection{}
+	for _, v := range s.Clients() {
+		v.proxy.RLock()
+		if v.proxy.WatchedResources[res.TypeUrl] != nil {
```

Comment:
nit: I think `v.Watching(res.TypeUrl)` handles the lock, nil check

Reply:
yeah.. good point. Fixed in https://github.com/istio/istio/pull/29740